### PR TITLE
Feature/mark post as read button

### LINKED ISSUE #384: A post action button to hide posts
Lemmy does have "Show/Hide read posts" but there are posts that you don't want to engage with at all. So, could it be possible to add a button to hide them from the UI so they don't appear on return visit? 

This will help user keep their front-page clean and might allow users to view more posts that interest them. As the clutter is be hidden.

May be a "Ignore" or "Hide" icon as a post action.

### PR DESCRIPTION
**Pull request by ItsEcholot**
Fixes #384

### CODE REVIEW

**Comment by SleeplessOne1917** [comment]
There are a bunch of changes from old commits showing up in this PR.

**Comment by ItsEcholot** [comment]
This is because I had to use branch release/0.17 while developing since the UI on main and BE on main and/or BE on 0.17 and UI on main aren’t compatible.
We can try to merge this into 0.17 if you want. Else I’m open to suggestions.

**Comment by SleeplessOne1917** [comment]
I'll try pulling this branch and merging main into it since the latest changes work on my machine.

**Comment by SleeplessOne1917** [comment]
@ItsEcholot I did a merge. Can you allow me to push changes to this branch?

**Comment by ItsEcholot** [comment]
@SleeplessOne1917 Sure, sent you an invite to the fork.

File: src/shared/components/community/create-community.tsx [hunk 1]
```diff
@@ -48,6 +50,7 @@ export class CreateCommunity extends Component<any, CreateCommunityState> {
   render() {
     return (
       <div className="container-lg">
+        {!UserService.Instance.myUserInfo && <Redirect to="/login" />}
```
Comment (SleeplessOne1917) [inline src/shared/components/community/create-community.tsx:53]: This logic is already taken care of with the `AuthGaurd` component.

File: src/shared/components/post/create-post.tsx [hunk 2]
```diff
@@ -143,6 +144,7 @@ export class CreatePost extends Component<
 
     return (
       <div className="container-lg">
+        {!UserService.Instance.myUserInfo && <Redirect to="/login" />}
```
Comment (SleeplessOne1917) [inline src/shared/components/post/create-post.tsx:147]: Taken care of by `AuthGuard`.

File: src/shared/components/post/post-listing.tsx [hunk 3]
```diff
@@ -750,6 +752,25 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
     );
   }
 
+  get markReadButton() {
+    const read = this.props.post_view.read;
+    const label = read ? i18n.t("mark read") : i18n.t("mark unread");
```
Comment (SleeplessOne1917) [inline src/shared/components/post/post-listing.tsx:757]: Please don't put spaces in the translation strings. Use `mark_read` and `mark_unread` instead.

File: src/shared/components/post/post-listing.tsx [hunk 4]
```diff
@@ -1497,6 +1518,20 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
...
+    const auth = myAuth();
+    if (auth) {
+      const read =
+        i.props.post_view.read == undefined ? true : !i.props.post_view.read;
+      const form: MarkPostAsRead = {
+        post_id: i.props.post_view.post.id,
+        read,
+        auth,
+      };
+      WebSocketService.Instance.send(wsClient.markPostAsRead(form));
```
Comment (SleeplessOne1917) [inline src/shared/components/post/post-listing.tsx:1531]: Was `markPostAsRead` added to the typescript client project? Also, was the work done on the backend to support this?

**Review by SleeplessOne1917 — COMMENTED** [review]
Besides the other comments I left, I'm gonna hold off on thinking about merging this until the websocket API removal has hit main.

File: src/shared/components/community/create-community.tsx (hunk 1 above)
Comment (ItsEcholot) [reply src/shared/components/community/create-community.tsx:53]: This is not part of my commit

File: src/shared/components/post/create-post.tsx (hunk 2 above)
Comment (ItsEcholot) [reply src/shared/components/post/create-post.tsx:147]: Same as above

File: src/shared/components/post/post-listing.tsx (hunk 4 above)
Comment (ItsEcholot) [reply src/shared/components/post/post-listing.tsx:1531]: Yes this is already implemented

File: src/shared/components/community/create-community.tsx (hunk 1 above)
Comment (SleeplessOne1917) [reply src/shared/components/community/create-community.tsx:53]: It must have snuck in during the merge.

File: package.json [hunk 5]
```diff
@@ -1,8 +1,7 @@
 {
   "name": "lemmy-ui",
-  "version": "0.17.1",
+  "version": "0.17.3",
   "description": "An isomorphic UI for lemmy",
-  "repository": "https://github.com/LemmyNet/lemmy-ui",
```
Comment (dessalines) [inline package.json:5]: Why did this get removed?

Comment (SleeplessOne1917) [reply package.json:5]: It could be another error that snuck in during the merge.

**Comment by SleeplessOne1917** [comment]
@ItsEcholot there are some merge conflicts that need handling before this can be merged.

**Comment by ItsEcholot** [comment]
I did some basic merging but didn't have time to test the functionality atm... If someone else wants to do some testing and if needed fixing it would be appreciated, else I'm gonna get around to it myself some time in the future.

**Comment by dessalines** [comment]
Closing as this is stale and needs a rebase.

**Pull request closed without merging**